Route/CmsPageRouter.php

```diff
@@ -207,7 +210,25 @@ protected function generateFromPage(PageInterface $page, array $parameters = arr
             throw new \RuntimeException(sprintf('Page "%d" has no url or customUrl.', $page->getId()));
         }
 
-        return $this->decorateUrl($url, $parameters, $referenceType);
+        if (!$this->context instanceof SiteRequestContext) {
+            @trigger_error(
+                'The context should implement SiteRequestContextInterface on the next release',
+                E_USER_DEPRECATED
+            );
+
+            return $this->decorateUrl($url, $parameters, $referenceType);
+        }
```

Comment:
I can see that `SiteRequestContext` does not implement this interface yet. If this means people will get a deprecation message every time without being able to do anything about it, this is not good. Deprecation messages should be thrown when and only when people can do something about it on the spot.